File: static/CWE-89/python/patched.py
# Remediation for CWE-89: SQL Injection
# Fix: Use a parameterized query so user input is bound as data, not concatenated into SQL.

import sqlite3
from flask import Flask, request

app = Flask(__name__)


@app.route("/user")
def get_user():
    user_id = request.args.get("id")
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE id = ?", (user_id,))
    return str(cursor.fetchall())

File: static/CWE-89/javascript/patched.js
// Remediation for CWE-89: SQL Injection
// Fix: Use a parameterized query with a placeholder so untrusted input is treated as data, not SQL.

const express = require('express');
const mysql = require('mysql');

const app = express();
const connection = mysql.createConnection({
  host: 'localhost',
  user: 'root',
  password: 'root',
  database: 'app',
});

app.get('/users/:id', (req, res) => {
  const userId = req.params.id;
  const query = 'SELECT * FROM users WHERE id = ?';
  connection.query(query, [userId], (err, results) => {
    if (err) return res.status(500).send('error');
    res.json(results);
  });
});

app.listen(3000);
